lucille-core/src/test/java/com/kmwllc/lucille/indexer/SolrIndexerTest.java

```diff
@@ -275,7 +296,176 @@ public void testSolrIndexerWithMultipleCollections() throws Exception {
     assertEquals("doc4", collectionDocs.get("col2").get(1).getFieldValue(Document.ID_FIELD));
     assertFalse(collectionDocs.get("col2").get(0).containsKey(indexOverrideField));
     assertFalse(collectionDocs.get("col2").get(1).containsKey(indexOverrideField));
+  }
+
+  @Test
+  public void testSolrIndexerDelete() throws Exception {
+    String deletionMarkerField = "is_deleted";
+    String deletionMarkerFieldValue = "true";
+
+    Config config =
+        ConfigFactory.empty()
+            .withValue("indexer.batchSize", ConfigValueFactory.fromAnyRef(1))
+            .withValue(
+                "indexer.deletionMarkerField", ConfigValueFactory.fromAnyRef(deletionMarkerField))
+            .withValue(
+                "indexer.deletionMarkerFieldValue",
+                ConfigValueFactory.fromAnyRef(deletionMarkerFieldValue));
+    PersistingLocalMessageManager manager = new PersistingLocalMessageManager();
+
+    Document doc1 = Document.create("doc1", "test_run");
+    doc1.addToField(deletionMarkerField, deletionMarkerFieldValue);
+
+    SolrClient solrClient = mock(SolrClient.class);
+    Indexer indexer = new SolrIndexer(config, manager, solrClient, "");
+    manager.sendCompleted(doc1);
+    indexer.run(1);
+
+    ArgumentCaptor<List<String>> delCaptor = ArgumentCaptor.forClass(List.class);
+
+    InOrder inOrder = inOrder(solrClient);
+
+    inOrder.verify(solrClient).deleteById(delCaptor.capture());
+    inOrder.verify(solrClient).close();
+
+    assertEquals(1, delCaptor.getAllValues().size());
+  }
+
+  @Test
+  public void testSolrIndexerAddDeleteAdd() throws Exception {
+    String deletionMarkerField = "is_deleted";
+    String deletionMarkerFieldValue = "true";
+
+    Config config =
+        ConfigFactory.empty()
+            .withValue("indexer.batchSize", ConfigValueFactory.fromAnyRef(3))
+            .withValue(
+                "indexer.deletionMarkerField", ConfigValueFactory.fromAnyRef(deletionMarkerField))
+            .withValue(
+                "indexer.deletionMarkerFieldValue",
+                ConfigValueFactory.fromAnyRef(deletionMarkerFieldValue));
+    PersistingLocalMessageManager manager = new PersistingLocalMessageManager();
+
+    Document doc1 = Document.create("doc1", "test_run");
+    doc1.addToField("version", 1);
+    Document doc2 = Document.create("doc1", "test_run");
+    doc2.addToField(deletionMarkerField, deletionMarkerFieldValue);
+    Document doc3 = Document.create("doc1", "test_run");
+    doc3.addToField("version", 2);
+
+    SolrClient solrClient = mock(SolrClient.class);
+    Indexer indexer = new SolrIndexer(config, manager, solrClient, "");
+    manager.sendCompleted(doc1);
+    manager.sendCompleted(doc2);
```

Comment:
It might be worth adding an unrelated add before doc1 and confirming that the both adds are sent before the deleteById... In other words, demonstrate that delete causes all pending adds to be processed first, whether the pending add is for the doc with ID "doc1" or something different.